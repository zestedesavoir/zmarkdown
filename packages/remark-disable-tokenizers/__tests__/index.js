import dedent from 'dedent'
import {unified} from 'unified'
import reParse from 'remark-parse'
import stringify from 'rehype-stringify'
import remark2rehype from 'remark-rehype'

import plugin from '../lib/index'

test('block', () => {
  const {value} = unified()
    .use(reParse)
    .use(plugin, [
      'blockQuote'
    ])
    .use(remark2rehype)
    .use(stringify)
    .processSync(dedent`
      *a*

      > *a*

      foo \`bar\` baz
    `)

  expect(value).toMatchSnapshot()
})

test('inline', () => {
  const {value} = unified()
    .use(reParse)
    .use(plugin, [
      'attention'
    ])
    .use(remark2rehype)
    .use(stringify)
    .processSync(dedent`
      *a*

      > *a*

      foo \`bar\` baz
    `)

  expect(value).toMatchSnapshot()
})

test('block + inline', () => {
  const {value} = unified()
    .use(reParse)
    .use(plugin, [
      'blockQuote',
      'attention'
    ])
    .use(remark2rehype)
    .use(stringify)
    .processSync(dedent`
      *a*

      > *a*

      foo \`bar\` baz
    `)

  expect(value).toMatchSnapshot()
})

test('does nothing', () => {
  const {value} = unified()
    .use(reParse)
    .use(plugin)
    .use(remark2rehype)
    .use(stringify)
    .processSync(dedent`
      *a*

      > *a*

      foo \`bar\` baz
    `)

  expect(value).toMatchSnapshot()
})

test('unknown tokenizer', () => {
  const {value} = unified()
    .use(reParse)
    .use(plugin, [
      'foo bar'
    ])
    .use(remark2rehype)
    .use(stringify)
    .processSync(dedent`
      *a*

      > *a*

      foo \`bar\` baz
    `)

  expect(value).toMatchSnapshot()
})
