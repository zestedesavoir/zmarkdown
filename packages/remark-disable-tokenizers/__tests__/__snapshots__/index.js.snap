// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`block + inline 1`] = `
"<p>*a*</p>
<p>> *a*</p>
<p>foo <code>bar</code> baz</p>"
`;

exports[`block 1`] = `
"<p><em>a</em></p>
<p>> <em>a</em></p>
<p>foo <code>bar</code> baz</p>"
`;

exports[`does nothing 1`] = `
"<p><em>a</em></p>
<blockquote>
<p><em>a</em></p>
</blockquote>
<p>foo <code>bar</code> baz</p>"
`;

exports[`inline 1`] = `
"<p>*a*</p>
<blockquote>
<p>*a*</p>
</blockquote>
<p>foo <code>bar</code> baz</p>"
`;

exports[`unknown tokenizer 1`] = `
"<p><em>a</em></p>
<blockquote>
<p><em>a</em></p>
</blockquote>
<p>foo <code>bar</code> baz</p>"
`;
